### DISCUSSION TITLE
select all content in Edit Box window with ctrl + A

### DISCUSSION THREAD

**Discussion by NgVThangBz** [axmolengine/axmol#2236] [General]
currently with EditBox, if you double click the entire content will be selected,
you can use ctrl + (C,V,X), but Ctrl + A doesn't work, 
is there any way to enable it,
or have any other ideas

**Comment by rh101** [comment]
Try the changes in PR #2238 to see if it works for you.

**Reply by NgVThangBz** [reply]
@rh101 
i pick this commit to my project (2.2.1),
it works fine on the first Editbox initialized, if there are more than 1 EditBox at the same time, it only works on the first EditBox initialized
![Screenshot 2024-11-25 at 13 43 14](https://github.com/user-attachments/assets/377f879b-00a5-484c-9f50-08401ad324fa)

**Reply by rh101** [reply]
I'll take a look at it today

**Reply by rh101** [reply]
The problem seems to be that the same hotkey combination cannot be registered more than once, even with a unique identifier, so it cannot be applied to more than one EditBox.
```
_hotKeyIdCtrlA = GlobalAddAtom(L"CTRL+A");
RegisterHotKey(_hwndEdit, _hotKeyIdCtrlA, MOD_CONTROL | MOD_NOREPEAT, 'A');
```

Even if this is changed to something like following to give a unique ID per key combination, it still does not work; `RegisterHotKey` returns `0`, and `GetLastError` returns ERROR_HOTKEY_ALREADY_REGISTERED:
```
auto uniqueId = fmt::format("CTRL+A|{}", fmt::ptr(this));
std::u16string utf16UniqueId;
ax::StringUtils::UTF8ToUTF16(uniqueId, utf16UniqueId);
_hotKeyIdCtrlA = GlobalAddAtom(LPCWSTR)utf16UniqueId.c_str());
RegisterHotKey(_hwndEdit, _hotKeyIdCtrlA, MOD_CONTROL | MOD_NOREPEAT, 'A');
```

A possible way to work around this is to register the hotkey for the cocos window, and then any time the hotkey `CTRL+A` is detected, we check if an EditBox is currently focused.  If it is not focused, then nothing happens, it ignores the hotkey combo.  If an EditBox is focused, then we apply the hotkey action to that EditBox.  This works in my tests, but, the one problem I can think of right now is that the `CTRL+A` hotkey would not work for anything else except the `EditBox`, which may not be desirable behaviour.

@halx99 Perhaps for the time being we revert PR #2238, and I'll create another PR with a different solution to this for review.  At the moment I can't think of any other solution to this besides what I mentioned above.

**Reply by rh101** [reply]
@NgVThangBz Try PR #2246

**Reply by NgVThangBz** [reply]
> @NgVThangBz Try PR #2246

@rh101 it doesn't seem to work for me (in all cases)
does it work on your side?

**Reply by rh101** [reply]
What do you mean by "all cases"? Can you show sample code?

If I create 3 instances of EditBox, then I can click on each and then use the `CTRL+A` to select the text in that EditBox.

**Reply by NgVThangBz** [reply]
```
    _textLeftWidth = EditBox::create(Size(20, 8), "png/input_background.png");
    _textLeftWidth->setPlaceHolder("Left");
    _textLeftWidth->setPlaceholderFontName(FONTS);
    _textLeftWidth->setFontName(FONTS);
    _textLeftWidth->setContentSize(Size(contentWidth, 14));
    _textLeftWidth->setSizeType(ui::Widget::SizeType::ABSOLUTE);
    _textLeftWidth->setFontColor(Color3B::BLACK);
    _textLeftWidth->setColor(cocos2d::Color3B::WHITE);
    _textLeftWidth->setInputMode(ui::EditBox::InputMode::NUMERIC);
    _textLeftWidth->setInputFlag(ui::EditBox::InputFlag::INITIAL_CAPS_SENTENCE);
    _textLeftWidth->setMaxLength(128);
    _textLeftWidth->setText("10000");
```
 this my code
even if there is 1 editbox, it doesn't work on my side

**Reply by rh101** [reply]
It's definitely working with your code.  Are you certain you have the code from PR #2246?

**Reply by rh101** [reply]
Here is a recording, with the bottom control being created from the code you posted:
![SelectAll](https://github.com/user-attachments/assets/98fc90f9-b1a9-4fac-8a84-0a386ac3d0b5)

**Reply by NgVThangBz** [reply]
@rh101   yes that was my mistake it works fine. sr
thank you

**Reply by rh101** [reply]
@NgVThangBz The current solution causes side-effects, so unfortunately we can't use this at all.  PR #2238 will also need to be reverted.  Maybe down the line we will find a better way to add this functionality.

**Reply by rh101** [reply]
@NgVThangBz Please try #2251 when you have a chance. It's a solution that should have no side-effects.

**Reply by NgVThangBz** [reply]
@rh101 yes, it work fine for me now, thank you!

**Comment by NgVThangBz** [comment]
wow it works fine thanks for your quick and good response

**Reply by aismann** [reply]
@NgVThangBz , if it working well, plz close it too.
